### ISSUE TITLE
Migrate log image location and use SAF

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#9609]
Currently c:geo is using `Environment.getExternalStoragePublicDirectory` to store log images the user takes or selects while writing his geocache logs. 
Details about the current disk usage structure can be found here: https://github.com/cgeo/cgeo/wiki/Disk-usage-structure
We are talking about item <4> of the table in this wiki article.

Starting with target SDK=30 Android enforces to use the StorageAccessFramework (SAF) instead of plain file access permissions.
Therefore we should migrate from the current access mode into using the SAF.

The general approach has been discussed in #8457 already, which I try to summarize below (correct me, if I am wrong) for this item:
- `getExternalStoragePublicDirectory` is deprecated anyway, so it needs to be replaced sooner or later
- The majority voted to therefore remove this method of storing images and use the same location as for external user data (see #9606)

To be agreed/discussed:
- Is migration needed or could we simply ignore what is stored in `/pictures/cgeo`?
- Does it make sense to remove the deprecated method even before introducing SAF (two-stepped-approach) or better do it in the same step?

**Comment by Lineflyer** [comment]
Migration is needed:
Taken from https://github.com/cgeo/cgeo/issues/8457#issuecomment-751467376

>Remember that log images' Uris are explicitly saved in the database. Once the base dir will turn to SAF we have no way to enforce a certain dir any more, user can choose whatever she wants and only this dir will have access by the user (today's dir cgeo/pictures has not). So at least a second migration step would be necessary once user selects cgeo base dir via SAF the first time. As I said, haven't thought this through myself but I want to prevent having to copy many images around each time user changes base dir...

**Comment by eddiemuc** [comment]
Partly adressed in branch `storage_framework`, please use it as a base for contributions.
Aslo please give me (@eddiemuc) a short hint before you do since I currently `rebase ` this branch onto `master ` frequently using forced pushes.

**Comment by eddiemuc** [comment]
@Lineflyer with regards to the open points in the description:

> Is migration needed or could we simply ignore what is stored in /pictures/cgeo?

If we don't migrate then all images for existing offline logs at time of release install are lost... 

> Does it make sense to remove the deprecated method even before introducing SAF (two-stepped-approach) or better do it in the same step?

There is nothing we can replace the deprecated method before moving to SAF. We need to store the images somewhere. And the deprecated method `Environment.getExternalStoragePublicDirectory()` has the advantage of still working in API30 (in contrast to the also deprecated `Environment.getExternalStorageDirectory()`, this will definitely no longer work)

**Comment by Lineflyer** [comment]
> If we don't migrate then all images for existing offline logs at time of release install are lost...

**Comment by eddiemuc** [comment]
> > There is nothing we can replace the deprecated method before moving to SAF. We need to store the images somewhere.
> 
> Couldn't we just move them from the current location (e.g. /Pictures/cgeo) to the legacy location we currently use (i.e. /cgeo/*)?

Sure, we can do that. But that would be something totally separate from the SAF migration because with API30 this legacy location will not work any more. Also for this move we would need to code a migration routine.

> That should be possible right now even before SAF integration. Wouldn't this mean, we can later skip file migration for pictures by just implementing #9606?

No, with #9606 we would then need to write a second migration for the log images. This is because of the special case that their file paths are stored in the database, and with #9606 we have to convert those to SAF content Uris. Log images are the only public content of c:geo where Uris are stored in the db

**Comment by eddiemuc** [comment]
PR #9734 will prepare master for SAF migration. The SAF mig itself will be done when storage_framework is merged into master

**Comment by eddiemuc** [comment]
Blocked by #9806
